tools/shared.py

```diff
@@ -2871,10 +2871,14 @@ def path_to_system_js_libraries(library_name):
 
   @staticmethod
   def emit_wasm_source_map(wasm_file, map_file):
+    # source file paths must be relative to the location of the map (which is
+    # emitted alongside the wasm)
+    base_path = os.path.dirname(os.path.abspath(Settings.WASM_BINARY_FILE))
```

Comment:
It probably doesn't matter since they're emitted alongside, but can't we use `map_file` here?

Reply:
sadly no; we run these commands on temp files. the new code here plugs through the final path of the output map file.

Reply:
Makes sense, thanks.